receiver/prometheusexec/README.md

```diff
@@ -0,0 +1,126 @@
+
+    prometheus_exec/postgresql:
+        exec: ./postgres_exporter
+        env:
+          - name: DATA_SOURCE_NAME
+            value: postgresql://user:password@url:port/
+
+exporters:
+    stackdriver:
+        project: "project_name"
+    file:
+        path: ./file_name.json
+
+service:
+    pipelines:
+        metrics:
+            receivers: [prometheus_exec/mysqld, prometheus_exec/postgresql]
+            exporters: [stackdriver, file]
+```
+
+## Config
+For every `prometheus_exec` defined in the configuration file, a command will be run (you *should* start a binary for this receiver to serve its purpose) and an equivalent Prometheus receiver will be instantiated to scrape its metrics.
+
+- ### prometheus_exec (required)
+`prometheus_exec` receivers should hierarchically be placed under the `receivers` key. You can define as many of these as you want, and they should be named as follows: `prometheus_exec/custom_name`. The `custom_name` should be unique per receiver you define, and is important for logging and error tracing. If no `custom_name` is given (i.e. simply having `prometheus_exec`), one will be generated for you by using the first word in the command to be executed (doing this limits the amount of receivers to 1 though since you can't define multiple receivers with duplicate names, so it's not recommended). Example:
+
+```yaml
+receivers:
+    prometheus_exec/mysql: 
+        # custom_name here is mysql
+
+    prometheus_exec:
+        exec: ./postgresql_exporter --port 1234 --config conf.xml 
+        # custom_name here is ./postgresql_exporter
+```
+
+- ### exec (required)
+Under each `prometheus_exec/custom_name` there needs to be an `exec` key. The value of this key is a string of the command to be run, with any flags needed (this will probably be the binary to run, in the correct relative directory). Environment variables can be set in a later configuration setting. Example:
+
+```yaml
+receivers:
+    prometheus_exec/apache:
+        exec: ./apache_exporter --retry=false
+
+    prometheus_exec/postgresql:
+        exec: ./postgresql_exporter --port 1234 --config conf.xml
+```
+
+- ### port
+`port` is an optional entry. Its value is a number indicating the port the receiver should be scraping the binary's metrics from. Two important notes about `port`:
+1. If it is omitted, we will try to randomly generate a port for you, and retry until we find one that is free. Beware when using this, since you also need to indicate your binary to listen on that same port with the use of a flag and string templating inside the command, which is covered in 2.
+
+2. **All** instances of `{{port}}` in any string of any key for the enclosing `prometheus_exec` will be replaced with either the port value indicated or the randomly generated one if no port value is set with the `port` key.
+
+Example:
+
+```yaml
+receivers:
+    prometheus_exec/apache:
+        exec: ./apache_exporter --retry=false
+        port: 1234 
```

Comment:
Can you make sure all the examples in this file are legitimate as people may copy/paste, e.g. looks like the default port for this exporter is [9117](https://github.com/Lusitaniae/apache_exporter/blob/8c2e8538a06c63ed247001fad9cd3a63d97ced97/apache_exporter.go#L28)

Reply:
Good point - I'll make them as real as possible! Thanks

Reply:
Good point, I'll make them as real as possible - thanks!